docs/extending/overview_extending.rst

```diff
@@ -0,0 +1,300 @@
+.. copied from edx-platform/blob/master/docs/guides/extension_points.rst
+
+###############################
+Extension options: an overview
+###############################
+
+.. contents::
+   :depth: 2
+   :local:
+
+************
+Introduction
+************
+
+Open edX platform development follows the `Open-Closed Principle`_: we offer an extensible platform that allows developers to build extensions that integrate with the core. This allows the core to remain small while volatile extensions remain in the periphery.
+
+As you will see in this document, there are many different ways to integrate with Open edX. However, we know that there are still some features and integrations that are not possible today without modifying the core. If you have such a need, please consider proposing a new extension point in the core that would make possible the functionality you have in mind. When you submit a pull request for a new extension point, be sure to include a change to this file to document your new extension point. (Doing so will also notify reviewers that want to help with making the platform more extensible.)
```

Comment:
> When you submit a pull request for a new extension point, be sure to include a change to this file to document your new extension point. (Doing so will also notify reviewers that want to help with making the platform more extensible.)

If this is moved out of `edx-platform`, this sentence will need to be re-worded to say "When you submit a pull request for a new extension point, **please open a simultaneous pull request** to this documentation to document your new extension point. (Doing so will notify..." and include a link to edit the .rst source file

Reply:
@robrap I'm going to assume that we're moving this file out of edx-platform so I'm going to go ahead with this suggested change.